cmd/antrea-agent/agent.go

```diff
@@ -750,6 +751,18 @@ func run(o *Options) error {
 
 	go agentMonitor.Run(stopCh)
 
+	var supportBundleController *support.SupportBundleController
+
+	if o.nodeType == config.ExternalNode {
+		supportBundleController = support.NewSupportBundleController(nodeConfig.Name, config.ExternalNode, o.config.ExternalNode.ExternalNodeNamespace, antreaClientProvider,
+			ovsctl.NewClient(o.config.OVSBridge), agentQuerier, networkPolicyController, v4Enabled, v6Enabled)
+	} else {
+		supportBundleController = support.NewSupportBundleController(nodeConfig.Name, config.K8sNode, "", antreaClientProvider,
+			ovsctl.NewClient(o.config.OVSBridge), agentQuerier, networkPolicyController, v4Enabled, v6Enabled)
+	}
+
+	go supportBundleController.Run(stopCh)
```

Comment:
This code will run unconditionally? I feel it needs a feature gate to isolate the code since it would affect K8s case too.
@jianjuns

Reply:
I have added a feature gate as suggested. Please help check the latest version.